src/integration-tests/test_subscriptions.py

```diff
@@ -1934,3 +1934,100 @@ def test_poison(cluster: Cluster, domain_urls: tc.DomainUrls):
     delayed = [producer.post(x=2)]  # pylint: disable=unused-variable
 
     consumer2.expect_messages(expected2, confirm=True)
+
+
+def test_stuck_downstream(multi_node: Cluster, domain_urls: tc.DomainUrls):
```

Comment:
I think this test should not belong to this file, because:
1. It doesn't touch subscription feature as we configure it during negotiation and doesn't test it. The only relation to subscriptions feature is the internal representation in the broker, so we threat all the consumers with the same appId and empty subscription as the same subscription group.
2. It doesn't depend or use high-level `Consumer`/`Producer` classes introduced for this series of tests, which suggests that it's not related.

I would suggest to move it to another file, somewhere where we test routing/delivery.